backend/backend/auth.py

```diff
@@ -0,0 +1,60 @@
+from typing import Annotated, Any
+
+import jwt
+from fastapi import Depends, HTTPException
+from fastapi.security import OpenIdConnect
+from sqlmodel import Session, select
+
+from backend.config import get_settings
+from backend.db import get_session
+from backend.models.tables import User, UserType
+
+oidc = OpenIdConnect(openIdConnectUrl=get_settings().keycloak_connection_string)
+
+
+def verify_token(raw_token: Annotated[str, Depends(oidc)]) -> dict[str, Any]:
+    token = raw_token.strip().split()[1]
+    jwks_client = jwt.PyJWKClient(get_settings().keycloak_jwks_uri)
```

Comment:
avoid re-fatching JWKS for every request, instantiate client once at module level